### ISSUE TITLE
"An exception occurred while executing 'INSERT INTO RoleActionRecords'" on MariaDB

### ISSUE THREAD

**Issue by nikoloutsa** [GOCDB/gocdb#125]
Hello , i have an error when i am trying to attache a new role 
```
An exception occurred while executing 'INSERT INTO RoleActionRecords (actionDate, updatedByUserId, updatedByUserPrinciple, roleId, rolePreStatus, roleNewStatus, roleTypeId, roleTypeName, roleTargetOwnedEntityId, roleTargetOwnedEntityType, roleUserId, roleUserPrinciple) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params 

QLSTATE[23000]: Integrity constraint violation: 1048 Column 'roleId' cannot be null
```

**Comment by nikoloutsa** [comment]
Yes.

Currently our deployment is using: mysql  Ver 15.1 Distrib 5.5.56-MariaDB, for Linux (x86_64) using readline 5.1

My workaround is to explicitly set the Role -> id to a default value since for a weird reason it is not set as it should (its an auto-increment column):

```
MariaDB [GOCDB5]> describe Roles;
+----------------+--------------+------+-----+---------+----------------+
| Field          | Type         | Null | Key | Default | Extra          |
+----------------+--------------+------+-----+---------+----------------+
| id             | int(11)      | NO   | PRI | NULL    | auto_increment |
```

Current fix:
```
lib/Doctrine/entities/Role.php line 90:

$this->id = 'DEFAULT'; 
```

**Comment by gregcorbett** [comment]
Hi @nikoloutsa, I take it that work around was applied to the database directly?

**Comment by ElliottKasoar** [comment]
In general, the availability of generated identifiers is not guaranteed until a successful flush:

> Generated entity identifiers / primary keys are guaranteed to be available after the next successful flush operation that involves the entity in question. You can not rely on a generated identifier to be available directly after invoking persist.

The default identifier generation strategy for MySQL/MariaDB is `IDENTITY`, rather than  `SEQUENCE` for Oracle. When using `IDENTITY`, values are not generated until insertion, leading to the error.

**Comment by gregcorbett** [comment]
This has been fixed by #276
